### ISSUE TITLE
purge/clear command

### ISSUE THREAD

**Issue by Tais993** [Together-Java/TJ-Bot#17]
# Issue

Whenever theres a spam bot, or something related, deleting messages by hand isn't exactly an option.
At the times the moderators aren't active or online, this means manual removal of their messages.

# Implementation

Undetermined, read the convo's

# Alternatives

Waiting for moderators to join the party

**Comment by Zabuzard** [comment]
I propose a different logic to the command, basically I would align it to the other commands (the ban functionality specifically), which just deletes all messages from the last day or so.

It would be simpler to use and I think it would cover the use case as well.

**Comment by Tais993** [comment]
This requires us to cache every single message for 1 day, it's possible but sounds overkill to me.

Or helpers receive ban permissions (which shouldn't happen imo)

**Comment by AzatothTheAwakened** [comment]
Rename it "purge" because thats usually how its called in other bots

**Comment by Tais993** [comment]
> 100 messages of those users
This either requires us to store all messages send, or we need to retrieve ... messages from each channel to find their messages.
Both are, not really a nice solution imo.

One feels like a waste of storage, and one will cause rate-limits.
If the user only send 99, and we still try to delete 100 the bot will keep retrieving messages.
So there needs to be a limit of when the bot stops retrieving.

For your info, bots can't do "from:Tijs#0002" to see all messages send by "Tijs#0002"
The search API is only for users, so these kind of commands are hard to properly make.

**Comment by Tais993** [comment]
Thanks Github for removing the 0's..
And since both you and Zabuzard mentioned caching messages.
We can think about this, but we need to set a limit on when messages will be removed.

Like, we will store user ID's and message ID's for 5 days?

I honestly don't know how much storage this takes, so I'm not sure how realistic this is.
We can just try it when the vps is ready, see #27 
After results from how much storage messages take we can decide further if everyone is alright with that?

**Comment by Zabuzard** [comment]
Blocked by #59

**Comment by d3vel0per** [comment]
Storage shouldnt be a big problem. The previous TJ bot already did this and the size was below 200megs(or may be little more but not very significant) for storing the asked metadata for last 30 days

**Comment by d3vel0per** [comment]
Also I am not very familiar with sqllite but most or databases have data compression mechanisms to reduce storage cost even further. If needed we can swap it out with other products as well

**Comment by Tais993** [comment]
Only issue now is whenever we will store the content, if yes we need to encrypt it.

**Comment by RealYusufIsmail** [comment]
I am working on the purge command and will be included in my next pr which also includes the warn command. I might open the pr in 2 weeks time. I will start working on the warn command next week.

**Comment by Zabuzard** [comment]
Closing in favor of duplicate #330 .
